openpype/hosts/max/plugins/publish/validate_renderpasses.py

```diff
@@ -0,0 +1,156 @@
+    """
+
+    order = ValidateContentsOrder
+    families = ["maxrender"]
+    hosts = ["max"]
+    label = "Validate Render Passes"
+    optional = True
+    actions = [RepairAction]
+
+    def process(self, instance):
+        if not self.is_active(instance.data):
+            return
+        invalid = self.get_invalid(instance)
+        if invalid:
+            bullet_point_invalid_statement = "\n".join(
+                f"- {err_type}: {filepath}" for err_type, filepath
+                in invalid
+            )
+            report = (
+                "Invalid render passes found.\n\n"
+                f"{bullet_point_invalid_statement}\n\n"
+                "You can use repair action to fix the invalid filepath."
+            )
+            raise PublishValidationError(
+                report, title="Invalid Render Passes")
+
+    @classmethod
+    def get_invalid(cls, instance):
+        """Function to get invalid beauty render outputs and
+        render elements.
+
+        1. Checking Render Output Folder matches the name of
+        the current Max Scene
+        e.g. The name of the current Max scene:
+                John_Doe.max
+            The expected render output directory:
+            {root[work]}/{project[name]}/{hierarchy}
+            /{asset}/work/{task[name]}/render/3dsmax/John_Doe/
+
+        2. Checking the image extension(s) of the render output(s)
+          does not match with image format in OP/AYON setting.
+        e.g. The current image format in the setting: png
+            The expected render outputs: Joe_Doe.png
+
+        3. Checking the filename of render element is not ended
+            with the name of render element from the 3dsMax Render
+            Element Manager.
+        e.g. The name of render element: RsCryptomatte
+            The expected filename: {InstanceName}_RsCryptomatte.png
+
+        Args:
+            instance (pyblish.api.Instance): instance
+            filename (str): filename of the Max scene
+
+        Returns:
+            list: list of invalid filename which doesn't match
+                with the project name
+        """
+        invalid = []
+        file = rt.maxFileName
+        filename, ext = os.path.splitext(file)
+        if filename not in rt.rendOutputFilename:
+            cls.log.error(
+                "Render output folder "
+                "doesn't match the max scene name! "
+            )
+            invalid_folder_name = os.path.dirname(
+                rt.rendOutputFilename).replace(
+                    "\\", "/").split("/")[-1]
+            invalid.append(("Invalid Render Output Folder",
+                            invalid_folder_name))
+        beauty_fname = os.path.basename(rt.rendOutputFilename)
+        ext = os.path.splitext(beauty_fname)[-1].lstrip(".")
+        invalid_image_format = cls.get_invalid_image_format(
+            cls, instance, ext)
+        invalid.extend(invalid_image_format)
+        renderer_class = get_current_renderer()
```

Comment:
Isn't the renderer usually 'collected' data in the instance?

Reply:
yes it is